### PR TITLE
refactor(decimal): refactor to typescript - FE-4871

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour

This PR refactors Decimal to TypeScript

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Decimal regression testing

### CODE REVIEW

File: src/components/decimal/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { default } from "./decimal.component";
+export type { DecimalProps, CustomEvent } from "./decimal.component";
```
Comment (robinzigmond) [inline src/components/decimal/index.ts:2]: Are we sure that the `CustomEvent` type needs to be exported? I'm probably missing something but if not we should only export the minimum that consumers need imo.

Comment (mkrds) [reply src/components/decimal/index.ts:2]: I think it will be useful for the consumers as they need to type the event in their onChange handlers.

Comment (robinzigmond) [reply src/components/decimal/index.ts:2]: Yes that makes sense, thanks. Happy for this to be left in.

**Review by robinzigmond — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by robinzigmond — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by robinzigmond — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by robinzigmond — APPROVED** [review]

**Review by robinzigmond — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**